packages/core/test/unit/data-types/arrays.test.ts

```diff
@@ -146,4 +146,24 @@ describe('DataTypes.ARRAY', () => {
       });
     }
   });
+
+  describe('clone', () => {
+    if (!dialect.supports.dataTypes.ARRAY) {
+      return;
+    }
+
+    it('should not throw if cloning within a scope', () => {
+      const Foo = sequelize.define('foo', {
+        bar: DataTypes.ARRAY(DataTypes.STRING(128)),
```

Comment:
You can call clone on the DataType directly, you don't need to use scopes in this test

Reply:
If I just run `DataTypes.ARRAY(DataTypes.STRING(128)).clone()` then it does not throw an error, though. Only in the context of scopes do I get the error...

Reply:
Yes but we know why: the instance of the data type inside of `DataTypes.ARRAY` must be different, and it did not use to be

The test can simply check that `array.options.type` is not the same instance on both arrays.

This isn't a bug specific to scopes. Any place that clones data types with the goal of using them on another model is going to fail, including model inheritance

Reply:
Alright, how about this?